staging/src/k8s.io/apimachinery/pkg/util/wait/loop.go

```diff
@@ -46,6 +46,15 @@ func loopConditionUntilContext(ctx context.Context, t Timer, immediate, sliding
 			return ctx.Err()
 		case <-timeCh:
 		}
+	} else {
```

Comment:
Add godoc to this function indicating the guarantee.

Reply:
Can we reverse the if block so that we do “if immediate {  } else { }” instead?  That will be idiomatic and slightly ease reading.